### PR TITLE
Add tracking for required client libraries for components

### PR DESCRIPTION
**Pull request by badvision**
In order to properly generate a form, client libraries should be included.  Most usually this means common libraries but there are some components which require special libraries.  In order to accommodate these, the FieldComponent and GeneratedDialog now handle client libraries so this can be surfaced more easily.

### CODE REVIEW

**Comment by davidjgonzalez** [comment]
@badvision also - is this PR intended to expose them? If so, I'm missing where they get written out in such a way that the clientlibs requested..?

**Comment by badvision** [comment]
@davidjgonzalez The GeneratedDialog class exposes them for use in the HTL.  Right now there is no front-end component for GeneratedDialog.  That's coming later, this is just ground-work.

File: bundle/src/main/java/com/adobe/acs/commons/mcp/form/FieldComponent.java [hunk 1]
```diff
@@ -110,6 +120,38 @@ public ResourceMetadata getComponentMetadata() {
...
+    public void addClientLibraries(ClientLibraryType type, String... categories) {
+        Set<String> categoriesSet = clientLibraries.getOrDefault(type, new LinkedHashSet<>());
+        for (String category : categories) {
+            categoriesSet.add(category);
+        }
+        clientLibraries.put(type, categoriesSet);
+    }
+
+    public void addClientLibraries(ClientLibraryType type, Collection<String> categories) {
+        Set<String> categoriesSet = clientLibraries.getOrDefault(type, new LinkedHashSet<>());
```
[the remark this answers is not in the document]
Comment (badvision) [reply bundle/src/main/java/com/adobe/acs/commons/mcp/form/FieldComponent.java]: Funny that it didn't but you have a point.

**Comment by davidjgonzalez** [comment]
Multifield makes sense (i had to write some custom JS for a granite multifield just the other day!)

I don't recall but are the generated elements well identified (so the JS can easily grab them?) ... iirc using typical granite UI field definitions you can pepper the XML w/ some granite:XXX properties that add data-* or classes to elements helping you identify them (though I'm not sure if that's an anti-pattern and CoralUI JS APIs should be used for everything, rather then DOM selection)

**Pull request merged**